packages/instantsearch.js/src/lib/InstantSearch.ts

```diff
@@ -203,6 +203,7 @@ class InstantSearch<
    * The error gets cleared when the next valid search response is rendered.
    */
   public error: Error | undefined = undefined;
+  public willRerender: boolean = false;
```

Comment:
I'm fine with this being a global variable, but I think this name doesn't sufficiently explain what/when it's used. A couple things on my mind:

- I don't want anyone outside of of InstantSearch depending on this, so it should be prefixed with `_`
- I'm not sure `re-render` is what we want to convey, maybe it's "unstable widgets", "pending render", "render requested" or something more like that?

Reply:
Totally makes sense. I'll schedule a quick sync so we can figure out the best way to approach this.

Reply:
Following our sync, I switched the approach to always delay setting the usertoken if it is received after init.